src/main/java/ch/jalu/typeresolver/EnumUtil.java

```diff
@@ -0,0 +1,99 @@
+package ch.jalu.typeresolver;
+
+import javax.annotation.Nullable;
+import java.util.EnumSet;
+import java.util.Optional;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+
+/**
+ * Utilities for enum classes.
+ */
+public final class EnumUtil {
+
+    private EnumUtil() {
+    }
+
+    /**
+     * Returns the enum entry with the given name if the provided class is an enum and has an entry that matches
+     * exactly the given name. Null is returned otherwise.
+     *
+     * @param clazz the class to check for enum entries, or null
+     * @param name the name to match, or null
+     * @param <T> the class type
+     * @return the enum entry if the name was non-null, the class was an enum and had such an enum entry; null otherwise
+     */
+    @Nullable
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    public static <T> T valueOfOrNull(@Nullable Class<T> clazz, @Nullable String name) {
```

Comment:
ClassUtil has a method `Optional<Class<?>> tryLoad(String name)`. I like "try". I don't necessarily like `Optional` much but I guess a decision should be made. This could be `Optional<T> tryValueOf(Class, String)`, which would work out quite nicely.